lnwallet/btcwallet/btcwallet.go

```diff
@@ -1285,6 +1285,50 @@ func getPreviousOutpoints(wireTx *wire.MsgTx,
 	return previousOutpoints
 }
 
+// GetTransactionDetails returns details of a transaction given its
+// transaction hash.
+func (b *BtcWallet) GetTransactionDetails(
+	txHash *chainhash.Hash) (*lnwallet.TransactionDetail, error) {
+
+	// Grab the best block the wallet knows of, we'll use this to calculate
+	// # of confirmations shortly below.
+	bestBlock := b.wallet.Manager.SyncedTo()
+	currentHeight := bestBlock.Height
+	tx, err := b.wallet.GetTransaction(*txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	// For both confirmed and unconfirmed transactions, create a
+	// TransactionDetail which re-packages the data returned by the base
+	// wallet.
+	if tx.Confirmations > 0 {
+		txDetails, err := minedTransactionsToDetails(
+			currentHeight,
+			base.Block{
+				Transactions: []base.TransactionSummary{
+					tx.Summary,
+				},
+				Hash:      tx.BlockHash,
+				Height:    tx.Height,
+				Timestamp: tx.Summary.Timestamp},
+			b.netParams,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		return txDetails[0], nil
+	}
+
+	txDetail, err := unminedTransactionsToDetail(tx.Summary, b.netParams)
+	if err != nil {
+		return nil, err
```

Comment:
nit: we could either add some additional error context here with `fmt.Errorf()`. If not, then I guess we can directly do `return unminedTransactionsToDetail(...)`.

Reply:
I will just return it then.